### DISCUSSION TITLE
请问心跳包我参数配置是否错误？为什么客户端老容易掉线，即使网络环境非常快？

### DISCUSSION THREAD

**Discussion by tg42** [kbengine/kbengine#1453] [Q&A]
我unity客户端clientapp里面的server heartbeat设置的是10秒，服务端<channelCommon>里面的<timeout> external也设置的是10秒，reliable udp里面的参数也配置得更大，默认值更容易掉线，<reliableUDP>
			
			<readPacketsQueueSize>
				<internal>				2048		</internal>
				<external>				2048			</external>
			</readPacketsQueueSize>

			
			<writePacketsQueueSize>
				<internal>				2048		</internal>
				<external>				2048			</external>
			</writePacketsQueueSize>

			
			<tickInterval>				10			</tickInterval>
			
			<minRTO>					10			</minRTO>
			
			<missAcksResend>			2			</missAcksResend>
			
			
			<mtu>						0			</mtu>

			
			<congestionControl>			false		</congestionControl>
			<nodelay>					true		</nodelay>
		</reliableUDP>

**Comment by zsj897** [comment]
我也遇到这样的问题，网络延迟一点，客户端就疯狂掉线。按理说哪怕有延迟，也只是发包慢，不应该掉线的。

**Comment by zsj897** [comment]
不过我是TCP。

**Comment by zsj897** [comment]
分享一下最新研究，unity sdk 接口有问题，会导致高网络延时 自己关闭socket , 文件PacketReceiverTCP.cs  bytesRead = socket.Receive(_buffer, _wpos, space, 0);  bytesRead可以为0。

**Reply by tg42** [reply]
请问你看了udp代码也有这个问题吗？
